.github/workflows/publish-ruby.yml

```diff
@@ -0,0 +1,96 @@
+name: Publish Ruby SDK
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  generate_schemas:
+    uses: ./.github/workflows/generate_schemas.yml
+
+  build_rust:
+    uses: ./.github/workflows/build-rust-cross-platform.yml
+
+  build_ruby:
+    name: Build Ruby
+    runs-on: ubuntu-22.04
+    needs:
+      - generate_schemas
+      - build_rust
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.2
+
+      - name: Download Ruby schemas artifact
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: schemas.rb
+          path: languages/ruby/lib
+
+      - name: Download x86_64-apple-darwin files
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: libbitwarden_c_files-x86_64-apple-darwin
+          path: temp/macos-x64
+
+      - name: Copy MacOS intel lib file
+        run: |
+          mkdir languages/ruby/lib/macos-x64
+          cp temp/macos-x64/libbitwarden_c.dylib languages/ruby/lib/macos-x64/libbitwarden_c.dylib
+        shell: bash
+
+      - name: Download aarch64-apple-darwin files
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: libbitwarden_c_files-aarch64-apple-darwin
+          path: temp/macos-arm64
+
+      - name: Copy MacOS arm lib file
+        run: |
+          mkdir languages/ruby/lib/macos-arm64
+          cp temp/macos-arm64/libbitwarden_c.dylib languages/ruby/lib/macos-arm64/libbitwarden_c.dylib
+        shell: bash
+
+      - name: Download x86_64-unknown-linux-gnu files
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: libbitwarden_c_files-x86_64-unknown-linux-gnu
+          path: temp/ubuntu-x64
+
+      - name: Copy Linux lib file
+        run: |
+          mkdir languages/ruby/lib/ubuntu-x64
+          cp temp/ubuntu-x64/libbitwarden_c.so languages/ruby/lib/ubuntu-x64/libbitwarden_c.so
+        shell: bash
+
+      - name: Download x86_64-pc-windows-msvc files
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: libbitwarden_c_files-x86_64-pc-windows-msvc
+          path: temp/windows-x64
+
+      - name: Copy Windows lib file
+        run: |
+          mkdir languages/ruby/lib/windows-x64
+          cp temp/windows-x64/bitwarden_c.dll languages/ruby/lib/windows-x64/bitwarden_c.dll
+        shell: bash
+
+      - name: Setup gem and release it
+        run: |
+          cd languages/ruby
+          gem install bundler
+          gem build *.gemspec
+          gem install *.gem
```

Comment:
The setup-ruby action should be able to configure bundler appropriately https://github.com/ruby/setup-ruby#bundler.

The install actions should be split from bundling it.